client/blocks/upsell-nudge/index.jsx

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'components/gridicon';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '@automattic/components';
+import TrackComponentView from 'lib/analytics/track-component-view';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export class UpsellNudge extends Component {
+	static defaultProps = {
+		className: '',
+	};
+
+	static propTypes = {
+		className: PropTypes.string,
+		eventName: PropTypes.string,
+		eventProperties: PropTypes.object,
+		buttonText: PropTypes.string,
+		icon: PropTypes.string,
+		href: PropTypes.string,
+		text: PropTypes.string,
+		onClick: PropTypes.func,
+		onDismissClick: PropTypes.func,
+		showDismiss: PropTypes.bool,
+		isCompact: PropTypes.bool,
+		track: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	onClick = e => {
+		const { eventName, eventProperties, track, onClick } = this.props;
+		track( 'calypso_upsell_nudge_button_click', { event_name: eventName, ...eventProperties } );
+		if ( onClick ) {
+			onClick( e );
+		}
+	};
+
+	onDismissClick = e => {
+		const { eventName, eventProperties, track, onDismissClick } = this.props;
+		track( 'calypso_upsell_nudge_button_dismiss_click', {
+			event_name: eventName,
+			...eventProperties,
+		} );
+		if ( onDismissClick ) {
+			onDismissClick( e );
+		}
+	};
+
+	render() {
+		const {
+			className,
+			eventName,
+			eventProperties,
+			buttonText,
+			href,
+			icon,
+			text,
+			isCompact,
+			showDismiss,
+			translate,
+		} = this.props;
+		const classes = classnames( 'upsell-nudge', className, isCompact ? 'compact' : '' );
+
+		return (
+			<div className={ classes }>
+				<TrackComponentView
```

Comment:
Might also be worth refactoring this in a separate janitorial, since we're looking at it. We can convert this to a function with a single [useEffect](https://reactjs.org/docs/hooks-reference.html#useeffect) and dispatch a single redux analytics event

In psuedocode it might looks something like
```jsx
function TrackComponentView () {
	const dispatch = useDispatch();

	useEffect( () => {
		dispatch( myAnaltyicsEvents() );
	}, [ dispatch ] );

	return null;
}
```